soroban-spec/src/gen/ts.rs

```diff
@@ -0,0 +1,339 @@
+use std::{fs, io};
+
+use super::json::types::{self, StructField, UnionCase};
+
+use heck::ToLowerCamelCase;
+use itertools::Itertools;
+use sha2::{Digest, Sha256};
+use stellar_xdr::ScSpecEntry;
+
+use types::Entry;
+
+use crate::{
+    gen::{json::types::Type, ts::wrapper::type_to_js_xdr},
+    read::{from_wasm, FromWasmError},
+};
+
+pub mod boilerplate;
+pub mod wrapper;
+
+#[derive(thiserror::Error, Debug)]
+pub enum GenerateFromFileError {
+    #[error("reading file: {0}")]
+    Io(io::Error),
+    #[error("sha256 does not match, expected: {expected}")]
+    VerifySha256 { expected: String },
+    #[error("parsing contract spec: {0}")]
+    Parse(stellar_xdr::Error),
+    #[error("getting contract spec: {0}")]
+    GetSpec(FromWasmError),
+}
+
+pub fn generate_from_file(
+    file: &str,
+    verify_sha256: Option<&str>,
+) -> Result<String, GenerateFromFileError> {
+    // Read file.
+    let wasm = fs::read(file).map_err(GenerateFromFileError::Io)?;
+
+    // Produce hash for file.
+    let sha256 = Sha256::digest(&wasm);
+    let sha256 = format!("{sha256:x}");
+
+    if let Some(verify_sha256) = verify_sha256 {
+        if verify_sha256 != sha256 {
+            return Err(GenerateFromFileError::VerifySha256 { expected: sha256 });
+        }
+    }
+
+    // Generate code.
+    let json = generate_from_wasm(&wasm).map_err(GenerateFromFileError::GetSpec)?;
+    Ok(json)
+}
+
+pub fn generate_from_wasm(wasm: &[u8]) -> Result<String, FromWasmError> {
+    let spec = from_wasm(wasm)?;
+    let json = generate(&spec);
+    Ok(json)
+}
+
+pub fn generate(spec: &[ScSpecEntry]) -> String {
+    let collected: Vec<_> = spec.iter().map(Entry::from).collect();
+    collected.iter().map(entry_to_ts).join("\n")
+}
+
+fn doc_to_ts_doc(doc: &str) -> String {
+    if doc.is_empty() {
+        String::new()
+    } else {
+        let doc = doc.split('\n').join("\n * ");
+        format!(
+            r#"/**
+ * {doc}
+ */
+"#,
+        )
+    }
+}
+
+pub fn entry_to_ts(entry: &Entry) -> String {
+    match entry {
+        Entry::Function {
+            doc,
+            name,
+            inputs,
+            outputs,
+        } => {
+            let sign_me = doc
+                .contains("@signme")
+                .then_some("sign: true, ")
```

Comment:
Is the plan to keep this? @chadoh If I understood some things you were saying today, it is no longer needed since preflight is used to determine when signing is required.

Reply:
Removing this!